hydra-node/test/Hydra/Model.hs

```diff
@@ -561,16 +577,16 @@ performNewTx party tx = do
       err@Output.TxInvalid{} -> error ("expected tx to be valid: " <> show err)
       _ -> False
 
-sendsInput :: Monad m => Party -> ClientInput Tx -> RunMonad m ()
+sendsInput :: (MonadSTM m, MonadThrow m) => Party -> ClientInput Tx -> RunMonad m ()
 sendsInput party command = do
   nodes <- gets nodes
   case Map.lookup party nodes of
-    Nothing -> error $ "unexpected party " <> Hydra.Prelude.show party
+    Nothing -> throwIO $ UnexpectedParty party
     Just actorNode -> lift $ actorNode `send` command
 
-performAbort :: MonadDelay m => Party -> RunMonad m ()
+performAbort :: (MonadDelay m, MonadSTM m, MonadThrow m) => Party -> RunMonad m ()
 performAbort party = do
-  Nodes{nodes} <- get
+  nodes <- gets nodes
```

Comment:
I liked the previous better. Also reading this on github makes me wonder.. is this not a naming conflict?